<launch>

<!-- Arguments -->
  <arg name="model" default="$(find mech_ros)/robot_description/urdf/mechROS.xacro"/>
  <arg name="gui" default="true" />
  <arg name="rvizconfig" default="$(find mech_ros)/rviz/navigation.rviz" />
  <arg name="map_file" default="$(find mech_ros)/map/muzeum_map.yaml"/>
  <arg name="prefix" default="mechROS_"/>
  
<!-- Global Parameters -->
  <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />
  <param name="use_gui" value="$(arg gui)"/>
  <rosparam file="$(find mech_ros)/control/mechROS_skid_steer_control.yaml" command="load"/> 



    <!-- IMU Madgwick fusion -->
  <node name="imu_filter_node" pkg="imu_filter_madgwick" type="imu_filter_node" output="screen">
    <param name="world_frame" value="enu" />
    <param name="use_magnetic_field_msg" value="false" />
    <param name="use_mag" value="true" />
    <param name="gain" value="0.08" />
    <param name="zeta" value="0.002" />
    <param name="orientation_stddev" value="0.2" />
    <param name="fixed_frame" value="imu_link" />
    <param name="publish_tf" value="false" />
        <!-- Debug -->
    <!-- <param name="publish_debug_topics" value="true" /> -->
    <!-- <param name="stateless" value="true" /> -->
  </node>

<!-- Correct orientation from Madgwick for declination -->
 <node name="declination_corrector" pkg="mech_ros" type="Declination_correction.py"/>

<!-- Simulated robot -->

<node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
output="screen" ns="/mechROS" args="base_control"> </node>


  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
      <param name="use_gui" value="False"/>
   </node>

   <node pkg="robot_state_publisher" type="robot_state_publisher"  name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
  </node>
 
 <!-- Rviz visualization -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

 
  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

<!-- Detect markers from image -->
 <node name="markers_detector" pkg="mech_ros" type="Aruco_Gazebo_detect.py"/>

    <!-- Estimate pose from marker -->
   <node pkg="mech_ros" type="Estimated_pose_markers.py"  name="marker_pose_estimator">
    <param name="markers_map" value="/home/patrik/catkin_ws/src/mech_ros/map/muzeum_aruco_markers.yaml" />
    <param name="min_covariance" value="0.0001" />
    <param name="marker_detector_topic" value="/markers" />
    <param name="estimated_pose_topic" value="/estimated_pose_markers" />
  </node>

 <!-- Extended Kalman - relatve -->
  <include file="$(find mech_ros)/launch/ext_kalman_relative.launch"/>

   <!-- Extended Kalman - absolute -->
  <include file="$(find mech_ros)/launch/ext_kalman_absolute.launch"/>

     <!-- Navigation -->
  <!-- <include file="$(find mech_ros)/launch/move_base_flex_simul.launch"/> -->

   <!-- Charging -->
<!-- <node pkg="mech_ros" type="charging.py" name="mbf_move_base_action" output="screen"/> -->

   <!-- Follow marker -->
<node pkg="mech_ros" type="Follow_target_marker.py" name="marker_follower" output="screen">
  <param name="cmd_vel_topic" value="/mechROS/base_control/cmd_vel" />
</node>

  
</launch>